### PR TITLE
Issue 19972 make it easier to set values for image file fields

### LINKED ISSUE #19972: Make it easier to set values for Image/File fields
### Is your feature request related to a problem? Please describe.

It's extremely tedious to set values for Image and File fields.  As reported by a customer: https://dotcms.zendesk.com/agent/tickets/103626

```
I find the Image Field uploading experience not convenient and I think the UI needs a little update.
There's one particular point that I can't stand every time I have to upload an image and it is that it is
not pointing by default to the host I am under in dotCMS. Since we need to upload the images in
the actual sites, then I think it's only normal to point to its directory right away.

Right now I have to go through the whole list of 100+ sites and open the folders manually.
Imagine uploading or updating 20 contentlet Images that way.

I have to reselect my host directory, open the little arrow to view the subfolders, click on asset and
select my image. This is a very long process.
```

### Describe the solution you'd like

There are several ways we could make this process (much) easier.  Here are my suggestions (in order of the value I think they'd have for customers):

1. Have dotCMS remember the last selected folder.
    * This might take a cookie, or might require some other info to be stored in the browser session.
        * But if the value stored was an identifier, keeping it in a cookie would present no privacy or security implications.
2. Create a "Favorites" (or equivalent) branch in the site selection tree.
    * Allow users to mark folders as favorites, to get them to show up in this list.
    * This could actually be the best overall solution (since it has value beyond just Image/File fields).
        * But it's probably a heavier lift than just remembering the last selected folder.
    * Note: This could possibly be done without too much work by adding "Favorite" as a type of Menu Link choice.
3. By default, select the current site (and open it in the navigation tree).
4. Allow specification of a default folder for Image fields and File fields.
    * Probably via a configuration property.

### Describe alternatives you've considered

* Image/File fields can not be imported or exported.
* So, the only real alternative right now appears to be to update the values using an external script making REST API calls.

### PR DESCRIPTION
**Pull request by freddyucv**

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/browser/BrowserUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,456 @@
+package com.dotmarketing.portlets.browser;
```
Comment (erickgonzalez) [inline dotCMS/src/integration-test/java/com/dotmarketing/portlets/browser/BrowserUtilTest.java:1]: Need to add this test to the MainSuite.class

File: dotCMS/src/main/java/com/dotmarketing/portlets/browser/BrowserUtil.java [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+
+public class BrowserUtil {
```
Comment (erickgonzalez) [inline dotCMS/src/main/java/com/dotmarketing/portlets/browser/BrowserUtil.java:35]: Add some doc. What is the idea of this Util? Add Javadoc to methods.

**Review by erickgonzalez — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/browser/BrowserUtil.java [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+
+    private static final String LAST_SELECTED_FOLDER_ID = "LAST_SELECTED_FOLDER_ID";
+    private static final String HOST_INDICATOR = "//";
+    final static List<DefaultPathResolver> defaultPathResolver;
+
+    static {
+        defaultPathResolver = list(
+                (contentlet, field, user) -> resolveWithCurrentValue(contentlet, field),
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/browser/BrowserUtil.java:46]: Very good

File: dotCMS/src/main/java/com/dotmarketing/portlets/browser/BrowserUtil.java [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+            } else {
+                throw  new NotFoundInDbException("Folder not found :" + folderId);
+            }
+        } catch (DotSecurityException | DotDataException e) {
+            throw new NotFoundInDbException("Folder not found :" + folderId, e);
+        }
+    }
+
+    @FunctionalInterface
+    interface DefaultPathResolver {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/browser/BrowserUtil.java:253]: Add doc

File: dotCMS/src/main/java/com/dotmarketing/portlets/browser/BrowserUtil.java (hunk 2 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotmarketing/portlets/browser/BrowserUtil.java:38]: done https://github.com/dotCMS/core/pull/21224/commits/4d0aadeff34b620c13f5c90ccf378cffc2960c7a

File: dotCMS/src/integration-test/java/com/dotmarketing/portlets/browser/BrowserUtilTest.java (hunk 1 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotmarketing/portlets/browser/BrowserUtilTest.java:1]: done https://github.com/dotCMS/core/pull/21224/commits/9785d32e2b7e3e0ba28f25ba0594735ebfb4379a

File: dotCMS/src/main/java/com/dotmarketing/portlets/browser/BrowserUtil.java (hunk 4 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotmarketing/portlets/browser/BrowserUtil.java:258]: done https://github.com/dotCMS/core/pull/21224/commits/99d5a93887f1bc8e191571995065e4a5c406a00c

**Review by erickgonzalez — APPROVED** [review]

**Review by jdotcms — APPROVED** [review]

**Pull request merged**